### ISSUE TITLE
[Datadog] API limits reached

### ISSUE THREAD

**Issue by bkamin29** [google/slo-generator#243]
Hello,

I'm facing a Datadog API rate limit issue when using the SLO Generator.
This problem seem due to a rate-limit implemented on Datadog APIs : https://docs.datadoghq.com/api/latest/rate-limits/

I don't know if from a SLO Generator point of view, we can be able to reduce our amount of API call to compute SLOs.

**Comment by lvaylet** [comment]
Hi @bkamin29,

Have you discussed Adeo' s architecture with @clement-pruvot?

Off the top of my head, they managed to generate an artifical rate limiting on the number of requests to the Datadoc API by calling the containerized SLO Generator (in Cloud Run) from an intermediate Cloud Function.

**Comment by ocervell** [comment]
Yes, that's what they did. We might help though by adding retries within the `slo-generator`, like we did for Dynatrace [here](https://github.com/google/slo-generator/blob/master/slo_generator/backends/dynatrace.py#L270).
